tutorials/AgileRL/agilerl_matd3.py

```diff
@@ -0,0 +1,222 @@
+from agilerl.hpo.tournament import TournamentSelection
+from agilerl.utils.utils import initialPopulation
+from tqdm import trange
+
+from pettingzoo.mpe import simple_speaker_listener_v4
+
+if __name__ == "__main__":
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    print("===== AgileRL Online Multi-Agent Demo =====")
+
+    # Define the network configuration
+    NET_CONFIG = {
+        "arch": "mlp",  # Network architecture
+        "h_size": [32, 32],  # Actor hidden size
+    }
+
+    # Define the initial hyperparameters
+    INIT_HP = {
+        "POPULATION_SIZE": 4,
+        "ALGO": "MATD3",  # Algorithm
+        # Swap image channels dimension from last to first [H, W, C] -> [C, H, W]
+        "CHANNELS_LAST": False,
+        "BATCH_SIZE": 32,  # Batch size
+        "LR": 0.01,  # Learning rate
+        "GAMMA": 0.95,  # Discount factor
+        "MEMORY_SIZE": 100000,  # Max memory buffer size
+        "LEARN_STEP": 5,  # Learning frequency
+        "TAU": 0.01,  # For soft update of target parameters
+        "POLICY_FREQ": 2,  # Policy frequnecy
+    }
+
+    # Define the simple speaker listener environment as a parallel environment
+    env = simple_speaker_listener_v4.parallel_env(continuous_actions=True)
+    env.reset()
+
+    # Configure the multi-agent algo input arguments
+    try:
+        state_dim = [env.observation_space(agent).n for agent in env.agents]
+        one_hot = True
+    except Exception:
+        state_dim = [env.observation_space(agent).shape for agent in env.agents]
+        one_hot = False
+    try:
+        action_dim = [env.action_space(agent).n for agent in env.agents]
+        INIT_HP["DISCRETE_ACTIONS"] = True
+        INIT_HP["MAX_ACTION"] = None
+        INIT_HP["MIN_ACTION"] = None
+    except Exception:
+        action_dim = [env.action_space(agent).shape[0] for agent in env.agents]
+        INIT_HP["DISCRETE_ACTIONS"] = False
+        INIT_HP["MAX_ACTION"] = [env.action_space(agent).high for agent in env.agents]
+        INIT_HP["MIN_ACTION"] = [env.action_space(agent).low for agent in env.agents]
+
+    # Not applicable to MPE environments, used when images are used for observations (Atari environments)
+    if INIT_HP["CHANNELS_LAST"]:
+        state_dim = [
+            (state_dim[2], state_dim[0], state_dim[1]) for state_dim in state_dim
+        ]
+
+    # Append number of agents and agent IDs to the initial hyperparameter dictionary
+    INIT_HP["N_AGENTS"] = env.num_agents
+    INIT_HP["AGENT_IDS"] = env.agents
+
+    # Create a population ready for evolutionary hyper-parameter optimisation
+    pop = initialPopulation(
+        INIT_HP["ALGO"],
+        state_dim,
+        action_dim,
+        one_hot,
+        NET_CONFIG,
+        INIT_HP,
+        population_size=INIT_HP["POPULATION_SIZE"],
+        device=device,
+    )
+
+    # Configure the multi-agent replay buffer
+    field_names = ["state", "action", "reward", "next_state", "done"]
```

Comment:
Same case as above.

Reply:
Yes, the algorithms work. We've benchmarked them against epymarl and we've seen good performance (which you can see on the agilerl github). In terms of the small change of done to termination in the field names list, unfortunately 'done' is needed in the replay buffer and so changing to 'termination' will need us to make adjustments to the MultiAgentReplayBuffer class.

Reply:
Ok, no worries then, I understand.